src/interfaces/ICozyLens.sol

```diff
@@ -44,76 +28,64 @@ interface ICozyLens {
 
   /// @notice Returns the weight of the market, identified by its `_set` and `_trigger` address, as a zoc.
-  function getMarketWeight(address _set, address _trigger) view external returns (uint256 _weight);
+  function getMarketWeight(ISet _set, address _trigger) view external returns (uint256 _weight);
 
   /// @notice Returns the address of the market's cost model, identified by its `_set` and `_trigger` address.
-  function getMarketCostModel(address _set, address _trigger) view external returns (address _costModel);
+  function getMarketCostModel(ISet _set, address _trigger) view external returns (ICostModel _costModel);
 
   /// @notice Returns the cost, excluding fees, to purchase `_protection` amount of protection in the market,
   /// identified by its `_set` and `_trigger` address. The cost is in units of the set's `asset`.
-  function getPreviewPurchaseDataCost(address _set, address _trigger, uint256 _protection) view external returns (uint256 _cost);
+  function getPreviewPurchaseDataCost(ISet _set, address _trigger, uint256 _protection) view external returns (uint256 _cost);
 
   /// @notice Returns the time remaining until the pending withdrawal with ID `_withdrawalId` in `_set` can be
   /// completed. The returned time is only equal to the amount of elapsed wall-clock time when the set is active, since
   /// time does not accrue during inactive states.
-  function remainingWithdrawalDelay(address _set, uint256 _withdrawalId) view external returns (uint256 _delay);
+  function remainingWithdrawalDelay(ISet _set, uint256 _withdrawalId) view external returns (uint256 _delay);
 
   /// @notice Returns the `SetConfig` for the `_set`.
-  function getSetConfig(address _set) view external returns (SetConfig memory);
+  function getSetConfig(ISet _set) view external returns (IConfig.SetConfig memory);
 
   /// @notice Returns the `MarketInfo` for the market, identified by its `_set` and `_trigger` address.
-  function getMarketInfo(address _set, address _trigger) view external returns (MarketInfo memory);
+  function getMarketInfo(ISet _set, address _trigger) view external returns (IConfig.MarketInfo memory);
 
   /// @notice Returns an array of all trigger addresses in the `_set`.
-  function allTriggers(address _set) view external returns (address[] memory _triggers);
-
-  // -------------------------------------
-  // -------- Address Computation --------
-  // -------------------------------------
-
-  /// @notice Given the `_salt`, `_asset`, `_setConfig`, and `_marketInfos` that will be used to create a set,
-  /// compute and return the address that set will be deployed to.
-  /// @dev The `_salt` is the user-provided salt, not the final salt after hashing with the chain ID.
-  function computeSetAddress(bytes32 _salt, address _asset, SetConfig memory _setConfig, SetConfig[] memory _marketInfos) view external returns (address);
```

Comment:
these two funcs have been removed from `CozyLens` and moved to the set and ptoken factories